mmv1/main.go

```diff
@@ -158,20 +158,27 @@ func main() {
 	})
 
 	var providerToGenerate provider.Provider
-	var productsForVersion []*api.Product
 
-	ch := make(chan string, len(allProductFiles))
+	productFileChannel := make(chan string, len(allProductFiles))
+	productsForVersionChannel := make(chan *api.Product, len(allProductFiles))
 	for _, pf := range allProductFiles {
-		ch <- pf
+		productFileChannel <- pf
 	}
 
 	for i := 0; i < len(allProductFiles); i++ {
 		wg.Add(1)
-		go GenerateProduct(ch, providerToGenerate, &productsForVersion, startTime, productsToGenerate, *resourceToGenerate, *overrideDirectory, generateCode, generateDocs)
+		go GenerateProduct(productFileChannel, providerToGenerate, productsForVersionChannel, startTime, productsToGenerate, *resourceToGenerate, *overrideDirectory, generateCode, generateDocs)
 	}
 	wg.Wait()
 
-	close(ch)
+	close(productFileChannel)
+	close(productsForVersionChannel)
+
+	var productsForVersion []*api.Product
+
+	for p := range productsForVersionChannel {
+		productsForVersion = append(productsForVersion, p)
+	}
 
 	slices.SortFunc(productsForVersion, func(p1, p2 *api.Product) int {
```

Comment:
When testing other changes, I had one occurrence of the Go engine crashing here with a nil pointer reference to `productsForVersion`. I immediately retried and it did not occur again. I believe this is due to hitting a rare race condition with the goroutines.

Using a channel instead of passing the array pointer directly to the goroutines should solve for this, but it is something to keep an eye out for.

Reply:
Good catch.

Reply:
Do we have a plan for if other like, not easily reproducible build engine errors occur within the following weeks-months?

Reply:
If we have more build engine errors relating to this block, we can remove the parallelization and let it run serially until we can fix the issue.